TodoApp/TodoApp.Api.Tests/ItemsControllerTests.cs

```diff
@@ -0,0 +1,128 @@
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(message.Content, Is.Not.Null);
+                Assert.That(message.TryGetContentValue<Item>(out var item), Is.Not.Null);
+                Assert.That(item.IsEqual(expected), Is.True);
+            });
+        }
+
+        [Test]
+        public async Task PutItemAsync_ReturnsNoContent()
+        {
+            const string id = "1";
+            var expected = ItemsController.Items[2];
+
+            var message = await ExecuteAsyncAction(async () => await _itemsController.PutItemAsync(id, expected));
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(message.Content, Is.Not.Null);
+                Assert.That(message.TryGetContentValue<Item>(out var item), Is.Not.Null);
+                Assert.That(item.IsEqual(expected), Is.True);
+            });
+        }
+
+        [Test]
+        public async Task DeleteItemAsync_ReturnsOkWithDeletedItem()
+        {
+            const string id = "3";
+            var expected = ItemsController.Items[3];
+
+            var message = await ExecuteAsyncAction(async () => await _itemsController.DeleteItemAsync(id));
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(message.Content, Is.Not.Null);
+                Assert.That(message.TryGetContentValue<Item>(out var item), Is.Not.Null);
+                Assert.That(item.IsEqual(expected), Is.True);
+            });
+        }
+
+        [Test]
+        public async Task PostItemAsync_ReturnsCreatedWithLinkToItem()
+        {
+            _itemsController.Request = new HttpRequestMessage
+            {
+                RequestUri = new Uri("http://localhost/api/v1/items")
+            };
+            _itemsController.Configuration.Routes.MapHttpRoute(
+                name: "PostNewItem",
+                routeTemplate: "api/test/{id}",
+                defaults: new { id = RouteParameter.Optional }
+            );
+
+            var newItem = ItemsController.Items[1];
+
+            var message = await ExecuteAsyncAction(async () => await _itemsController.PostItemAsync(newItem));
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.Created));
+                Assert.That(message.Headers.Location.AbsoluteUri, Is.EqualTo("http://localhost/api/test/2"));
+                Assert.That(message.Content, Is.Not.Null);
+                Assert.That(message.TryGetContentValue<Item>(out var item), Is.Not.Null);
+                Assert.That(item.IsEqual(newItem), Is.True);
+            });
+        }
+
+        private static async Task<HttpResponseMessage> ExecuteAsyncAction(Func<Task<IHttpActionResult>> httpAction)
```

Comment:
Very well done! 👏 

Now, move this to new file as an extension for controllers (only visible in tests!).